### PR TITLE
RavenDB-15518 Deserialization of 'HubName' when creating Sink Task

### PR DESCRIPTION
**Pull request by ayende**

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/Replication/PullReplicationAsSink.cs [hunk 1]
```diff
@@ -10,9 +10,12 @@ public class PullReplicationAsSink : ExternalReplicationBase
         public string CertificatePassword;
 
         [Obsolete("PullReplicationAsSink.HubDefinitionName is not supported anymore. Will be removed in next major version of the product. Use HubName instead.")]
-        public string HubDefinitionName { get => HubName; set => HubName = value; }
+        public string HubDefinitionName { get; set; }
 
-        public string HubName;
+        private string _hubName;
+#pragma warning disable 618 // disable use obselete property
+        public string HubName { get => _hubName ?? HubDefinitionName; set => _hubName = value; }
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/Replication/PullReplicationAsSink.cs:17]: Maybe it is better to use 'IFillFromBlittableJson' or 'IPostJsonDeserialization' ?

Comment (ayende) [reply src/Raven.Client/Documents/Operations/Replication/PullReplicationAsSink.cs:17]: The problem happens during _serialization_, not deserialization, though.

Comment (ppekrol) [reply src/Raven.Client/Documents/Operations/Replication/PullReplicationAsSink.cs:17]: Then maybe we want to do:
```
public string HubName { get => _hubName ?? (_hubName = HubDefinitionName); set => _hubName = value; }
```
?

Comment (ayende) [reply src/Raven.Client/Documents/Operations/Replication/PullReplicationAsSink.cs:17]: Why does that matter?

Comment (ppekrol) [reply src/Raven.Client/Documents/Operations/Replication/PullReplicationAsSink.cs:17]: Hmm, actually now when I'm thinking about this, it doesn't.

**Pull request merged**